pkg/virt-handler/vm.go

```diff
@@ -535,7 +535,7 @@ func (d *VirtualMachineController) processVmShutdown(vmi *v1.VirtualMachineInsta
 	if clientDisconnected == false {
 
 		// Only attempt to gracefully shutdown if the domain has the ACPI feature enabled
-		if domain.Spec.Features.ACPI != nil {
+		if domain.Spec.Features != nil && domain.Spec.Features.ACPI != nil {
```

Comment:
You will also have to check for `ACPI.Enabled`. The feature is enabled if `Enabled` is unset or `true`.

Reply:
so there is a Enabled field missing here? https://github.com/slintes/kubevirt/blob/9d1dd9432b11dda136eb99e5133b6fb54383f09f/pkg/virt-launcher/virtwrap/api/schema.go#L161

Reply:
Oh sorry I was in the wrong schema. Everything good! :+1: